docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
+
+![Help Command Sequence Diagram](./images/HelpCommand_Sequence.png)
+
+#### Add Command
+
+An add command can be used to add a lesson to the timetable or add a module mapping to the user university list.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of an add command.
+
+![Add Command Sequence Diagram](./images/AddCommand_Sequence.png)
+
+#### Delete Command
+
+A delete command can be used to delete a lesson from the timetable, delete a module mapping from the user university list or delete an entire user created university list.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a delete command.
+
+![Delete Command Sequence Diagram](./images/DeleteCommand_Sequence.png)
+
+#### View Command
+
+A view command can be used to view all user created university lists, view user's delete history, view user's selected university list or view all the user's created university lists' timetables.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a view command.
+
+![View Command Sequence Diagram](./images/ViewCommand_Sequence.png)
+
+#### List Command
+
+A list command can be used to display all the module mappings in the database, all the universities in the database, or allow users to filter by NUS module code or partner university name.
+
+The following class diagram illustrates the relationship between the respective classes involved in the creation and execution of a list command.
+
+![List Command Class Diagram](./images/ListCommand_Class.png)
+
+To differentiate between the various functions of the list command, a variable `listOption` is used. This variable signals Duke to execute the relevant commands accordingly.
+
+The following sequence diagram illustrates the flow of the program to read in the user input, parse the user input, check if it is a valid list command, and execute the relevant list command based on the `listOption`.
+
+![List Command Sequence Diagram](./images/ListCommand_Sequence.png)
+
+#### Favourite Command
+
+A favourite command can be used by the user to favourite or un-favourite lists of module mappings that they have curated. It can also be used to display all of the user's favourite lists and the module mappings they contain.
+
+To differentiate between the various functions of the favourite command, a variable `favouriteOption` is used. This variable signals Duke to execute the relevant commands accordingly.
+
+The following class diagram illustrates the relationship between the respective classes involved in the creation and execution of a favourite command.
+
+![Favourite Command Class Diagram](./images/FavouriteCommand_Class.png)
```

Comment:
As we are not taught to use the circle bubbles in the diagrams, which I think it really makes the diagram more clear, you do however might want to consider taking the bubbles away?